### PR TITLE
Fix ICE #68058

### LINKED ISSUE #68058: Macro ICE
This code 
```rust
macro_rules! def_target {
    ($target: expr) => {
        #[target_feature(enable=$target)]
        unsafe fn f() {
            #[target_feature(enable=$target)]
            ()
        }
    };
}

def_target!("avx2");

fn main() {}

```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b147ca0ca4cda9fcf59142ba19a0be30))

Produces ICE on any channel:

```
   Compiling playground v0.0.1 (/playground)
thread 'rustc' panicked at 'src/librustc/ich/impls_syntax.rs:326: interpolated tokens should not be present in the HIR', src/librustc/util/bug.rs:37:26
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.40.0 (73528e339 2019-12-16) running on x86_64-unknown-linux-gnu

note: compiler flags: -C codegen-units=1 -C debuginfo=2 --crate-type bin

note: some of the compiler flags provided by cargo are hidden

error: could not compile `playground`.
```

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #68058

r? @petrochenkov

### CODE REVIEW

**Comment by petrochenkov** [comment]
Thanks!
@bors r+ rollup

File: src/librustc_ast_lowering/expr.rs [hunk 1]
```diff
@@ -202,7 +202,12 @@ impl<'hir> LoweringContext<'_, 'hir> {
             ExprKind::Mac(_) => panic!("Shouldn't exist here"),
         };
 
-        hir::Expr { hir_id: self.lower_node_id(e.id), kind, span: e.span, attrs: e.attrs.clone() }
+        hir::Expr {
+            hir_id: self.lower_node_id(e.id),
+            kind,
+            span: e.span,
+            attrs: e.attrs.iter().map(|a| self.lower_attr(a)).collect::<Vec<_>>().into(),
```
Comment (Centril) [inline src/librustc_ast_lowering/expr.rs]: Shouldn't this be using `attrs: self.lower_attrs(&e.attrs),`?

cc @Zoxc

Comment (Zoxc) [reply src/librustc_ast_lowering/expr.rs]: No, this uses `ThinVec`.

Comment (Centril) [reply src/librustc_ast_lowering/expr.rs]: Ah.

**Comment by JohnTitor** [comment]
Failed in https://github.com/rust-lang/rust/pull/68094#issuecomment-573115765
@bors r-

File: src/test/ui/macros/issue-68058.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    ($target: expr) => {
+        #[target_feature(enable=$target)]
+        unsafe fn f() {
+            #[target_feature(enable=$target)]
+            ()
+        }
+    };
+}
+
+def_target!("avx2");
```
Comment (JohnTitor) [inline src/test/ui/macros/issue-68058.rs]: @petrochenkov `// ignore-cloudabi no std::env` is enough to disable test on the unexpected environment?

File: src/test/ui/macros/issue-68058.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// check-pass
+
+macro_rules! def_target {
+    ($target: expr) => {
+        #[target_feature(enable=$target)]
```
Comment (petrochenkov) [inline src/test/ui/macros/issue-68058.rs]: This attribute is not relevant to the issue and can be removed.

Comment (petrochenkov) [reply src/test/ui/macros/issue-68058.rs]: `unsafe` on the function is irrelevant as well.

File: src/test/ui/macros/issue-68058.rs (hunk 2 above)
Comment (petrochenkov) [reply src/test/ui/macros/issue-68058.rs]: I don't think so.
Any target-neutral attribute (e.g. `#[doc = $expr]`) can be used instead, then these issues won't arise.

Comment (JohnTitor) [reply src/test/ui/macros/issue-68058.rs]: Got it, fixed test case.

File: src/test/ui/macros/issue-68058.rs (hunk 3 above)
Comment (JohnTitor) [reply src/test/ui/macros/issue-68058.rs]: Removed.

**Pull request merged**